src/OpenTelemetry.Collector.Dependencies/Implementation/AzureSdkDiagnosticListener.cs

```diff
@@ -119,6 +123,8 @@ public override void OnException(Activity current, object valueValue)
             }
 
             span.Status = Status.Unknown.WithDescription(valueValue?.ToString());
+
+            // todo: Determine if we need to call span.End() here or if OnStopActivity is also called for exceptions.
```

Comment:
Did you resolve this TODO?

Reply:
@MikeGoldsmith Hard to say 100% because there's a lot of code in the Azure SDK. I just did some testing and updated the comment from a "TODO" to a more conclusive "we're not calling span.End() on purpose" message. Here's what I did:

I tested the exception path using Azure.Security.KeyVault.Secrets.SecretClient. It wrote the Exception event and then the Stop event. Looking at the code it seems to be using a class DiagnosticScope to manage its events:
https://github.com/Azure/azure-sdk-for-net/blob/7dafccc43eb24f99bf135752b7d8db59d65d53da/sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs#L83-L100

DiagnosticScope calls Stop in its Dispose:
https://github.com/Azure/azure-sdk-for-net/blob/0eb893cfda1a9b39801d414ec77ce753eec43868/sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs#L72-L87

Assuming the whole SDK uses this pattern, we should be good.

Reply:
CC: @pakrym

Reply:
Yes, we still call Stop after the exception and the DiagnosticScope is what we are using in most places.

Reply:
@pakrym would you please send PR to replace `TODO: ` with the links to Azure SDK with the call to `Stop` or a comment about this. It will be really nice as this is inconsistent with how Http and Sql works.

Reply:
There is already a note https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry.Collector.Dependencies/Implementation/AzureSdkDiagnosticListener.cs#L127